tests/topotests/bgp_l3vpn_label_export/test_bgp_l3vpn_label_export.py

```diff
@@ -0,0 +1,587 @@
+                },
+            ]
+        }
+    elif label and rd:
+        expected = {
+            "102:1": {
+                "prefix": "192.0.3.2/32",
+                "paths": [
+                    {
+                        "valid": True,
+                        "remoteLabel": label,
+                        "nexthops": [{"ip": "0.0.0.0"}],
+                    }
+                ],
+            }
+        }
+    else:
+        expected = {}
+
+    return topotest.json_cmp(output, expected, exact=(label is None))
+
+
+def check_mpls_table(label, protocol):
+    tgen = get_topogen()
+
+    if label == "auto":
+        cmd = "show mpls table json"
+    else:
+        cmd = "show mpls table {} json".format(label)
+
+    output = json.loads(tgen.gears["r2"].vtysh_cmd(cmd))
+
+    if label == "auto" and protocol:
+        output_copy = deepcopy(output)
+        for key, data in output_copy.items():
+            for nexthop in data.get("nexthops", []):
+                if nexthop.get("type", None) != protocol:
+                    continue
+                output = data
+                break
+
+    if protocol:
+        expected = {
+            "nexthops": [
+                {
+                    "type": protocol,
+                },
+            ]
+        }
+    else:
+        expected = {}
+
+    return topotest.json_cmp(output, expected, exact=(protocol is None))
+
+
+def check_mpls_ldp_binding():
+    tgen = get_topogen()
+
+    output = json.loads(
+        tgen.gears["r1"].vtysh_cmd("show mpls ldp binding 192.0.2.2/32 json")
+    )
+    expected = {
+        "bindings": [
+            {
+                "prefix": "192.0.2.2/32",
+                "localLabel": "16",  # first available label
+                "inUse": 1,
+            },
+        ]
+    }
+
+    return topotest.json_cmp(output, expected)
+
+
+def test_convergence():
+    "Test protocol convergence"
+
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    step("Check BGP and LDP convergence")
+    test_func = functools.partial(check_bgp_vpn_prefix, 2222)
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, "Failed to see BGP prefix on R1"
+
+    test_func = functools.partial(check_mpls_ldp_binding)
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, "Failed to see LDP binding on R2"
+
+    test_func = functools.partial(check_mpls_table, 16, "LDP")
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, "Failed to see LDP label on R2"
+
+    test_func = functools.partial(check_mpls_table, 2222, "BGP")
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, "Failed to see BGP label on R2"
+
+    output = tgen.net["r2"].cmd("vtysh -c 'show debugging label-table' | grep Proto")
```

Comment:
We check once without any retries here? Dunno how it would work under a high load system (CI).

Reply:
if mpls table is updated, then show debugging label-table is updated.